src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/LicenceOverrideController.kt

```diff
@@ -0,0 +1,72 @@
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseBody
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.request.OverrideLicenceStatusRequest
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.LicenceOverrideService
+import javax.validation.Valid
+
+@RestController
+@RequestMapping("/licence/id/{licenceId}/override", produces = [MediaType.APPLICATION_JSON_VALUE])
+class LicenceOverrideController(private val licenceOverrideService: LicenceOverrideService) {
+  @PostMapping(value = ["/status"])
+  @PreAuthorize("hasAnyRole('SYSTEM_USER', 'CVL_ADMIN')")
+  @ResponseBody
+  @ResponseStatus(HttpStatus.ACCEPTED)
+  @Operation(
+    summary = "Override a licence status",
+    description = "Override the status for an exising licence. Only to be used in exceptional circumstances." +
+      " Requires ROLE_SYSTEM_USER or ROLE_CVL_ADMIN.",
+    security = [SecurityRequirement(name = "ROLE_SYSTEM_USER"), SecurityRequirement(name = "ROLE_CVL_ADMIN")],
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "202",
+        description = "Status has been updated"
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Bad request, request body must be valid",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorised, requires a valid Oauth2 token",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires an appropriate role",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "The licence for this ID was not found.",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      )
+    ]
+  )
+  fun changeStatus(
+    @PathVariable("licenceId") licenceId: Long,
+    @RequestBody @Valid request: OverrideLicenceStatusRequest
+  ) {
+    val licence = licenceOverrideService.getLicenceById(licenceId)
+    licenceOverrideService.changeStatus(licence!!, request.statusCode, request.reason)
```

Comment:
calling licenceOverrideService twice. Do as one step in the service?

Reply:
It could be within the same method. Having two methods ensures a separation of concerns, but requires two calls on the controller side (or a call to the method within the same service). Its largely subjective to a point. @andrewrlee any preference in this instance? 

The existing LicenceService is huge and often has the same code copied and pasted within each method to get the Licence.

Reply:
preference on one method call - then you don't need to expose a public `getLicenceById` method on the override service but not a big deal

A massive plus one on having a separate service!